R/get-net-task.R

```diff
@@ -1,19 +1,28 @@
-#' Returns an instance of the specified .NET Task
+#' Get an Instance of the Specified `.NET` Task
 #'
-#' @param taskName The name of the task to retrieve (without the Get)
+#' @param taskName The name of the task to retrieve (**without** `Get` prefix).
 #'
-#' @return An instance of the Task
+#' @return
+#'
+#' - `getNetTask()` returns an instance of of the specified `.NET` task.
+#' - `getContainerTask()` returns a reference to the container task for
+#' optimization purposes only.
+#'
+#' @examples
+#'
+#' if (FALSE) {
+#'   simulationLoader <- getNetTask("SimulationLoader")
+#' }
+#'
+#' ospsuite:::getContainerTask()
 #'
-#' @details
-#' simulationLoader <- getNetTask("SimulationLoader")
 #' @keywords internal
 getNetTask <- function(taskName) {
   rClr::clrCallStatic("OSPSuite.R.Api", paste0("Get", taskName))
 }
 
+#' @rdname getNetTask
+#' @keywords internal
 getContainerTask <- function() {
-  if (is.null(ospsuiteEnv$containerTask)) {
-    ospsuiteEnv$containerTask <- getNetTask("ContainerTask")
-  }
-  ospsuiteEnv$containerTask
+    ospsuiteEnv$containerTask %||% getNetTask("ContainerTask")
```

Comment:
This is not equivalent. You are removign setting in the environment.!

Reply:
Yikes! How come nothing fails?!

Reply:
This is an optimization. All the code will still work. Just a bit slower

Reply:
The idea is to call `getNetTaks("ContainerTask") only once, the instance is then cashed and returned from the ospsuiteEnv instead of creating a new one.

Reply:
You can add a test and make sure that this is set after it's being called. But the success of the test might depend on the order of how the tests are run etc... Not as easy as it seems.

Reply:
Reverted the change. I will see if I can think of a way to test this. But, yeah, non-trivial to test something like this.

One possibility is to use `{touchstone}`: https://lorenzwalthert.github.io/touchstone/

I'll explore that option if I find time.

Reply:
I not even sure you'll be able to measure a net difference with the few amounts of call we are doing.